packages/astro/src/core/build/vite-plugin-analyzer.ts

```diff
@@ -82,7 +82,7 @@ export function vitePluginAnalyzer(internals: BuildInternals): VitePlugin {
 				const astro = info.meta.astro as AstroPluginMetadata['astro'];
 
 				for (const c of astro.hydratedComponents) {
-					const rid = c.resolvedPath ? resolveClientDevPath(c.resolvedPath) : c.specifier;
+					const rid = c.resolvedPath ?? c.specifier;
```

Comment:
`resolvedPath` has already called `resolveClientDevPath` earlier, so we remove here.